changelog/all_assert_message.dd

```diff
@@ -0,0 +1,8 @@
+Better static assert messages for `std.algorithm.searching.all`
+
+Until now, `all` used a template constraint to check if the passed types
+could be used. If they were not, it was very tedious to figure out why.
+
+As the template constraint is not used for overload resolution
+the constrains are moved into static asserts with expressive error
+messages.
```

Comment:
If constraints are not necessary, I think you actually get more useful errors without static asserts, at least for `pred`. When `pred` is `a => a.len`:
```
find.d(9): Error: no property `len` for `a` of type `int`
/usr/include/dmd/phobos/std/algorithm/searching.d(1869):        instantiated from here: `not!int`
find.d(11):        instantiated from here: `find!(not, int[])`
```
Using a static assert hides the `len` typo. So I wouldn't say 'expressive' error messages.